scripts/generate.py

```diff
@@ -641,23 +650,25 @@ def data(self):
         try:
             data_container = self._data
         except AttributeError:
-            self._data = DataContainer()
-            return self._data
-        else:
-            # One more check in case the
-            # property setter is by-passed
-            if not isinstance(data_container, DataContainer):
-                raise TypeError("data is not a DataContainer. "
-                                "data.setter is by-passed.")
-            return data_container''')
+            self._data = DataContainer.new_with_restricted_keys(
+                self.supported_parameters())
+            data_container = self._data
+
+        retvalue = DataContainer.new_with_restricted_keys(
+            self.supported_parameters()
+            )
+        retvalue.update(data_container)
+
+        return retvalue''')
 
         self.methods.append('''
     @data.setter
     def data(self, new_data):
-        if isinstance(new_data, DataContainer):
-            self._data = new_data
-        else:
-            self._data = DataContainer(new_data)''')
+        data = DataContainer.new_with_restricted_keys(
```

Comment:
It is still valid. We want to leave it flexible, however. To the extend of CUBA keys.